### PR TITLE
fix(DB/Creature/SmartScript) Wood Mites should skitter around

### PR DESCRIPTION
**Pull request by M4th3m4tic4l**
## Changes Proposed:
- Causes the Wood Mites to start moving when spawning
- Move Rotting Forest-Rager script to SAI
- Information taken from: https://github.com/TrinityCore/TrinityCore/commit/4d338b1d4d6ddf449e52c38f6cb9cdeb9cbf32fe by Andrew Letson [velinath@users.noreply.github.com](mailto:velinath@users.noreply.github.com)

## Issues Addressed:

- https://github.com/azerothcore/azerothcore-wotlk/issues/14587
- https://github.com/chromiecraft/chromiecraft/issues/4625

## SOURCE:

https://www.youtube.com/watch?v=tMsU3BtEEfA&t=41s

## Tests Performed:

- 
WoWScrnShot_013023_231125

- 
WoWScrnShot_013023_231143

## How to Test the Changes:

1. Insert the changes in your world database and in src/server/scripts/Outland/zone_terokkar_forest.cpp
2. Compile and then do .go c 77579
3. Kill a creature

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: data/sql/updates/pending_db_world/rev_1675139356539586000.sql [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+(22095, 0, 2, 0, 6, 0, 100, 0, 0, 0, 0, 0, 0, 89, 5, 0, 0, 0, 0, 0, 11, 22419, 20, 1, 0, 0, 0, 0, 0, 'Infested Root-Walker - On Just Died - Start Random Movement');
+
+UPDATE `creature_template` SET `AIName` = 'SmartAI' WHERE `entry` = 22307;
+
+DELETE FROM `smart_scripts` WHERE (`entryorguid` = 22307) AND (`source_type` = 0) AND (`id` IN (0, 1, 2));
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
+(22307, 0, 0, 0, 6, 0, 75, 0, 0, 0, 0, 0, 0, 11, 39134, 2, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Rotting Forest-Rager - On Just Died - Cast \'Summon Lots of Wood Mights\''),
```
Comment (heyitsbench) [inline data/sql/updates/pending_db_world/rev_1675139356539586000.sql:11]: `Mights` -> `Mites`

Comment (M4th3m4tic4l) [reply data/sql/updates/pending_db_world/rev_1675139356539586000.sql:11]: funny

**Comment by Maelthyrr** [comment]
For the cherry-pick or PR based on another project, it's better to add information to the commit messages. Like this:

*Chery-pick: link
*Co-authored-by: name <name@users.noreply.github.com>

An example: https://github.com/azerothcore/azerothcore-wotlk/pull/12705/commits/70f282c25332440a7325cda245723bf753b45fde

**Comment by Maelthyrr** [comment]
Tested and work. Wood mites spawn and are not static. Thanks :)

File: data/sql/updates/pending_db_world/rev_1675139356539586000.sql [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
```
Comment (Nyeriah) [inline data/sql/updates/pending_db_world/rev_1675139356539586000.sql:13]:
```suggestion
(22307, 0, 0, 0, 6, 0, 75, 0, 0, 0, 0, 0, 0, 11, 39134, 2, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Rotting Forest-Rager - On Just Died - Cast \'Summon Lots of Wood Mites\''),
```

**Comment by Mitradis** [comment]
conflicts

**Review by Nyeriah — APPROVED** [review]

**Comment by Kitzunu** [comment]
Conflicts

File: data/sql/updates/pending_db_world/rev_1675139356539586000.sql (hunk 2 above)
Comment (heyitsbench) [reply data/sql/updates/pending_db_world/rev_1675139356539586000.sql]: [Blizzlike typo, btw](https://wowgaming.altervista.org/aowow/?spell=39134).

**Comment by FrancescoBorzi** [comment]
@M4th3m4tic4l thanks for this contribution. Can you please fix the merge conflicts so we can merge the PR?

**Comment by Kitzunu** [comment]
merge conflict

**Comment by pangolp** [comment]
image
Hello good evening. I managed to resolve the conflict, I would like to be able to upload it, but when I do so it gives me access denied and that's fine. The problem is that I can't push the branch to update this pull request and try to solve the problem.

Anyway, looking at the conflict, the person in this pull request made a SmartAI and there is another person who made a C++ script. We would have to do with any of the 2 scripts we stayed.

**Comment by pangolp** [comment]
Sorry, I read it wrong. I already solved the conflict. Lastly, I'll give you the file so that you can upload it. In the end, there are not 2 scripts, there is only one, the SmartAI.

**Comment by pangolp** [comment]
```diff
diff --git a/data/sql/updates/pending_db_world/rev_1675139356539586000.sql b/data/sql/updates/pending_db_world/rev_1675139356539586000.sql
new file mode 100644
index 000000000..a15cc064f
--- /dev/null
+++ b/data/sql/updates/pending_db_world/rev_1675139356539586000.sql
@@ -0,0 +1,15 @@
+--
+
+UPDATE `creature_template` SET `ScriptName` = "" WHERE `entry` = 22307;
+
+DELETE FROM `smart_scripts` WHERE (`entryorguid` = 22095) AND (`source_type` = 0) AND (`id` IN (2));
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
+(22095, 0, 2, 0, 6, 0, 100, 0, 0, 0, 0, 0, 0, 89, 5, 0, 0, 0, 0, 0, 11, 22419, 20, 1, 0, 0, 0, 0, 0, 'Infested Root-Walker - On Just Died - Start Random Movement');
+
+UPDATE `creature_template` SET `AIName` = 'SmartAI' WHERE `entry` = 22307;
+
+DELETE FROM `smart_scripts` WHERE (`entryorguid` = 22307) AND (`source_type` = 0) AND (`id` IN (0, 1, 2));
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
+(22307, 0, 0, 0, 6, 0, 75, 0, 0, 0, 0, 0, 0, 11, 39134, 2, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Rotting Forest-Rager - On Just Died - Cast \'Summon Lots of Wood Mites\''),
+(22307, 0, 1, 0, 0, 0, 100, 0, 0, 5, 12000, 18000, 0, 11, 15548, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Rotting Forest-Rager - In Combat - Cast \'Thunderclap\' (No Repeat)'),
+(22307, 0, 2, 0, 6, 0, 100, 0, 0, 0, 0, 0, 0, 89, 5, 0, 0, 0, 0, 0, 11, 22419, 20, 1, 0, 0, 0, 0, 0, 'Rotting Forest-Rager - On Just Died - Start Random Movement');
diff --git a/src/server/scripts/Outland/zone_terokkar_forest.cpp b/src/server/scripts/Outland/zone_terokkar_forest.cpp
index bc9f2ca24..e584f8cbe 100644
--- a/src/server/scripts/Outland/zone_terokkar_forest.cpp
+++ b/src/server/scripts/Outland/zone_terokkar_forest.cpp
@@ -429,38 +429,6 @@ public:
     };
 };
 
-/*######
-## npc_rotting_forest_rager
-######*/
-
-class npc_rotting_forest_rager : public CreatureScript
-{
-public:
-    npc_rotting_forest_rager() : CreatureScript("npc_rotting_forest_rager") { }
-
-    CreatureAI* GetAI(Creature* creature) const override
-    {
-        return new npc_rotting_forest_ragerAI(creature);
-    }
-
-    struct npc_rotting_forest_ragerAI : public ScriptedAI
-    {
-        npc_rotting_forest_ragerAI(Creature* creature) : ScriptedAI(creature) { }
-
-        void Reset() override { }
-        void JustEngagedWith(Unit* /*who*/) override { }
-
-        void DamageTaken(Unit* done_by, uint32& damage, DamageEffectType, SpellSchoolMask) override
-        {
-            if (done_by && done_by->GetTypeId() == TYPEID_PLAYER)
-                if (me->GetHealth() <= damage)
-                    if (rand() % 100 < 75)
-                        //Summon Lots of Wood Mights
-                        DoCast(me, 39134, true);
-        }
-    };
-};
-
 /*######
 ## npc_isla_starmane
 ######*/
@@ -714,7 +682,6 @@ void AddSC_terokkar_forest()
 
     // Theirs
     new npc_unkor_the_ruthless();
-    new npc_rotting_forest_rager();
     new npc_isla_starmane();
     new go_skull_pile();
     new npc_slim();
```

**Comment by pangolp** [comment]
Merge it before @M4th3m4tic4l touches anything else haha.
Lie, great contribution mate.
To renege less, and to move on.

**Comment by pangolp** [comment]
Next time remember:

```sql
UPDATE `creature_template` SET `AIName` = 'SmartAI' WHERE `entry` = 22307;
UPDATE `creature_template` SET `ScriptName` = "" WHERE `entry` = 22307;
```

and

```sql
UPDATE `creature_template` SET `AIName`='SmartAI', `ScriptName`="" WHERE `entry`=22307;
```

"It's the same", but one less query is one less query.
And with the number of inquiries that are sometimes made, 1 month is a lot.

**Pull request merged**